### PR TITLE
Psycopg dictionary results

### PR DESCRIPTION
**Pull request by elpavel**
##### SUMMARY
Modifying all psycopg results to use dictionary keys, and casting all ACL type data as varchar. 

This will ensure the same code will work with Psycopg 3, helping with #499. 

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
Many modules

##### ADDITIONAL INFORMATION
These changes were tested with Psycopg 3 in #517.

### CODE REVIEW

**Review by Andersson007 — APPROVED** [review]
@elpavel i like the gradual approach you use.
If we're sure it's backwards compatible and will not touch user workflows, this refactoring LGTM
thanks

File: plugins/modules/postgresql_info.py [hunk 1]
```diff
@@ -731,22 +731,22 @@ def get_tablespaces(self):
                               "AND column_name = 'spcoptions'")
 
         if not opt:
-            query = ("SELECT s.spcname, pg_catalog.pg_get_userbyid(s.spcowner) as rolname, s.spcacl "
+            query = ("SELECT s.spcname, pg_catalog.pg_get_userbyid(s.spcowner) as rolname, s.spcacl::varchar "
```
Comment (hunleyd) [inline plugins/modules/postgresql_info.py:734]: why varchar and not text? the field is of type aclitem and in pg varchar is just text w/ input limits

Comment (hunleyd) [reply plugins/modules/postgresql_info.py:734]: it probably doesn't make a difference, i'm just curious.

**Review by hunleyd — CHANGES_REQUESTED** [review]

File: plugins/modules/postgresql_info.py (hunk 1 above)
Comment (elpavel) [reply plugins/modules/postgresql_info.py:734]: I wasn't sure which one to use, whether text or varchar, so picked varchar by habit. 
Happy to change it, shouldn't make a difference.

**Review by Andersson007 — APPROVED** [review]

**Pull request merged**